### ISSUE TITLE
Flame fails when tool installed with pnpm

### ISSUE THREAD

**Issue by jsumners** [nearform/node-clinic#127]
## Expected Behavior
```
clinic flame -- node script.js # run test and ctrl+c should generate a flame graph
```

## Current Behavior
`ctrl+c` returns error if `clinic` was installed with pnpm:

```
[==  ] Analysing dataevents.js:183
      throw er; // Unhandled 'error' event
      ^

Error: ENOTDIR: not a directory, stat '/Users/jsumners/n/pnpm-global/1/node_modules/.registry.npmjs.org/clinic/3.0.0/node_modules/clinic/bin.js/node_modules/d3-dispatch'
```

## Steps to Reproduce (for bugs)
1. `npm i -g pnpm`
2. `pnpm i -g clinic`
3. `clinic flame -- node script.js`
4. `ctrl+c` to finish test and attempt to generate graph

## Environment

* Clinic version: v3.0.0

* Node version: v8.15.0

* Browser name and version: 

* Operating system and version: macOS 10.14.3

**Comment by AlanSl** [comment]
Thanks for reporting this, I can confirm that I get the same error in Node 11 as you do in Node 8.

I suspect it's because `d3-dispatch` is an implicit dep of a package it's required into. I get this issue in Clinic Flame only, not Bubbleprof or Doctor, and only with `d3-dispatch`. 

As a short term workaround, `pnpm i -g d3-dispatch` gets Clinic Flame working on pnpm like the other tools.

For a longer term fix, I've posted a PR on d3-fg to fix the implicit dependency: https://github.com/davidmarkclements/d3-fg/pull/31 - there doesn't seem to be anything else unusual about d3-dispatch.

**Comment by AlanSl** [comment]
@goto-bus-stop I see you've done some work on fixing Browserify stack paths for things like pnpm e.g. https://github.com/browserify/module-deps/pull/133 - do you think a required dep being implicit and not in package.json seems like an adequate explanation for why the browserify / module-deps stack might give an incorrect path on pnpm but not npm?

**Comment by jsumners** [comment]
Yes, I am sure that is the problem.

**Comment by AlanSl** [comment]
Appears fixed by d3-fg 6.13.2 release
